internal/alb/sg/lb_attachment.go

```diff
@@ -0,0 +1,119 @@
+package sg
+
+import (
+	"fmt"
+
+	"github.com/kubernetes-sigs/aws-alb-ingress-controller/internal/aws/albec2"
+
+	"github.com/aws/aws-sdk-go/aws"
+	"github.com/aws/aws-sdk-go/service/elbv2"
+
+	"github.com/kubernetes-sigs/aws-alb-ingress-controller/internal/aws/albelbv2"
+	"github.com/kubernetes-sigs/aws-alb-ingress-controller/pkg/util/log"
+)
+
+// LbAttachment represents the desired SecurityGroups attached to Lb
+type LbAttachment struct {
+	GroupIDs []string
+	LbArn    string
+}
+
+// LbAttachmentController controls the LbAttachment
+type LbAttachmentController interface {
+	// Reconcile ensures `only specified SecurityGroups` exists in LoadBalancer.
+	Reconcile(*LbAttachment) error
+
+	// Delete ensures specified SecurityGroup don't exists in LoadBalancer, other sg are kept.
+	// If there are remaining sg, the default SG for vpc will be kept.
+	Delete(*LbAttachment) error
+}
+
+type lbAttachmentController struct {
+	elbv2  albelbv2.ELBV2API
+	ec2    *albec2.EC2
+	logger *log.Logger
```

Comment:
What's the reason why each controller has its own logger?

Reply:
actually i don't need them, since golang's log already contains the file information, do you have any recommendations like use global logger?

Reply:
there should be a logger created for each ingress resource and then its been saved in the dependent structs for ease of use. the ingress logger includes the ingress name and namespace when printing.

Reply:
yeah, there could be a way like passing a reconcileContext into reconcile options. or have each object contains that reconcile context like ingress name.

Will leave it as it is and change it in later PR when needed.